tests/fips/gnutls/gnutls_server.pm

```diff
@@ -0,0 +1,83 @@
+#
+# Summary: SLES15SP2 FIPS certification, we need to certify gnutls and libnettle
+#          In this case, will configure GnuTLS server
+# Maintainer: rfan1 <richard.fan@suse.com>
+# Tags: poo#63223, tc#1744099
+
+use base "consoletest";
+use testapi;
+use strict;
+use warnings;
+
+sub run {
+    select_console "root-console";
+
+    # Create test folder
+    my $test_dir = "gnutls";
+    assert_script_run "mkdir $test_dir && cd $test_dir";
+
+    # Add support for X.509.  First we generate a CA
+    my $pri_ca_key = "x509-ca-key.pem";
+    my $ca_tmpl    = "ca.tmpl";
+    my $pri_ca_out = "x509-ca.pem";
+    assert_script_run "certtool --generate-privkey > $pri_ca_key";
+    assert_script_run "echo \"cn = GnuTLS test CA\"> $ca_tmpl";
+    assert_script_run "echo \"ca\" >> $ca_tmpl";
+    assert_script_run "echo \"cert_signing_key\" >> $ca_tmpl";
+    assert_script_run "certtool --generate-self-signed --load-privkey $pri_ca_key --template $ca_tmpl --outfile $pri_ca_out";
+
+    # Generate a server certificate
+    my $ser_pri_key = "x509-server-key.pem";
+    my $ser_tmpl    = "server.tmpl";
+    my $ser_ca_out  = "x509-server.pem";
+    assert_script_run "certtool --generate-privkey > $ser_pri_key";
+    assert_script_run "echo \"organization = GnuTLS test server\" > $ser_tmpl";
+    assert_script_run "echo \"cn = test.gnutls.org\" >> $ser_tmpl";
+    assert_script_run "echo \"tls_www_server\" >> $ser_tmpl";
+    assert_script_run "echo \"encryption_key\" >> $ser_tmpl";
+    assert_script_run "echo \"signing_key\" >> $ser_tmpl";
+    assert_script_run "echo \"dns_name = test.gnutls.org\" >> $ser_tmpl";
+    assert_script_run
+"certtool --generate-certificate --load-privkey $ser_pri_key --load-ca-certificate $pri_ca_out --load-ca-privkey $pri_ca_key --template $ser_tmpl --outfile $ser_ca_out";
+
+    #  Generate a client certificate
+    my $cli_pri_key = "x509-client-key.pem";
+    my $cli_tmpl    = "client.tmpl";
+    my $cli_out     = "x509-client.pem";
+    assert_script_run "certtool --generate-privkey > $cli_pri_key";
+    assert_script_run "echo \"cn = GnuTLS test client\" > $cli_tmpl";
+    assert_script_run "echo \"tls_www_client\" >> $cli_tmpl";
+    assert_script_run "echo \"encryption_key\" >> $cli_tmpl";
+    assert_script_run "echo \"signing_key\" >> $cli_tmpl";
+    assert_script_run
+"certtool --generate-certificate --load-privkey $cli_pri_key --load-ca-certificate $pri_ca_out --load-ca-privkey $pri_ca_key --template $cli_tmpl --outfile $cli_out";
+
+    # Create password file with psktool
+    my $user   = "psk_identity";
+    my $passwd = "psk-passwd.txt";
+    assert_script_run "psktool -u $user -p $passwd";
+
+    # Start a server with support for PSK. This would require a password file created with psktool
+    type_string "nohup gnutls-serv --http --priority NORMAL:+ECDHE-PSK:+PSK --pskpasswd $passwd&";
+    type_string "\n";
+}
+
+sub test_flags {
+    return {milestone => 1, fatal => 1};
+}
+
+1;
```

Comment:
There are lots of `"echo $string >> $file"`, it is not graceful
Please refer to e.g., "./Tomcat/Utils.pm" to concise the code.

Reply:
Thanks, will do

Reply:
Another question. 
There are lots of differences between code and https://bugzilla.suse.com/tr_show_case.cgi?case_id=1744099.
Any special considerations?

Reply:
Thanks, the steps mentioned in **https://bugzilla.suse.com/tr_show_case.cgi?case_id=1744099. ** is out of date, I will update with detail steps later. please refer to  https://progress.opensuse.org/issues/63223 for more detail info

Reply:
> Thanks, will do

done